### PR TITLE
 [Flink AI Flow] Add Serde of ContextExtractor to persist it in aiflow database with Workflow

### PR DESCRIPTION
**Pull request by bgeng777**
## What is the purpose of the change

Add Serde of ContextExtractor to persist it in aiflow database with Workflow

## Brief change log

- Use cloudpickle to do serde of ContextExtractor instance.
- Add a new field in WorkflowMeta in `message.proto`: `context_extractor_in_bytes`, whose type is `bytes`. It stores the serialized bytes of ContextExtractor instance.
- Modify the metadata service to adapt to the CRUD of `context_extractor_in_bytes` field.
- Add tests for above change

## Verifying this change

This change added tests.

### CODE REVIEW

File: flink-ai-flow/ai_flow/ai_graph/ai_graph.py [hunk 1]
```diff
@@ -36,7 +38,7 @@ class AIGraph(Graph):
     def __init__(self) -> None:
         super().__init__()
         self.nodes: Dict[Text, AINode] = {}
-        self._context_extractor: ContextExtractor = BroadcastAllContextExtractor()
+        self._context_extractor: bytes = cloudpickle.dumps(BroadcastAllContextExtractor())
```
Comment (Sxnan) [inline flink-ai-flow/ai_flow/ai_graph/ai_graph.py:41]: Though we had a talk about this offline, I am not sure why we should store it as bytes in AIGraph.

**Review by Sxnan — CHANGES_REQUESTED** [review]
Thanks for the patch. It looks good overall. I left some common below. One major concern is the place we serialize and deserialize the context extractor.

File: flink-ai-flow/examples/demo/workflows/simple_workflow/simple_workflow.py [hunk 2]
```diff
@@ -1,7 +1,37 @@
...
+import cloudpickle
+from notification_service.base_notification import BaseEvent
+
+from ai_flow.api.context_extractor import ContextExtractor, EventContext
+
 import ai_flow as af
 from ai_flow_plugins.job_plugins.bash import BashProcessor
 
 
+class TestContext(EventContext):
```
[the remark this answers is not in the document]
Comment (bgeng777) [reply flink-ai-flow/examples/demo/workflows/simple_workflow/simple_workflow.py:12]: Sure. I create a new workflow in demo.

File: flink-ai-flow/ai_flow/ai_graph/ai_graph.py (hunk 1 above)
Comment (bgeng777) [reply flink-ai-flow/ai_flow/ai_graph/ai_graph.py:41]: At first, I want to make it easier to serialize AIGraph in json_utils when building workflow. But I think it is not necessary to serialize the extractor in workflow.

File: flink-ai-flow/ai_flow/meta/workflow_meta.py [hunk 3]
```diff
@@ -30,6 +30,7 @@ def __init__(self,
                  properties: Properties = None,
                  create_time: int = None,
                  update_time: int = None,
+                 context_extractor_in_bytes: bytes = None,
```
[the remark this answers is not in the document]
Comment (bgeng777) [reply flink-ai-flow/ai_flow/meta/workflow_meta.py:37]: Add getter and setter which returns/accepts a ContextExtractor instance. 
But for the type of field, I think using bytes can reduce unnecessary serde.

**Review by Sxnan — APPROVED** [review]
LGTM

**Pull request merged**